Use clang++ instead of clang to create dylib.
Specify C++ stdlib with -stdlib flag not -l flag.
Use archs from Xcode rather than hardcoding.
--- MoltenVK/scripts/create_dylib_macos.sh.orig	2018-09-08 18:45:39.000000000 -0500
+++ MoltenVK/scripts/create_dylib_macos.sh	2018-09-12 20:03:10.000000000 -0500
@@ -8,9 +8,9 @@
 export MVK_SYS_FWK_DIR="${SDK_DIR}/System/Library/Frameworks"
 export MVK_USR_LIB_DIR="${SDK_DIR}/usr/lib"
 
-clang \
+clang++ \
 -dynamiclib \
--arch x86_64 \
+$(printf " -arch %s" ${ARCHS}) \
 -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET} \
 -compatibility_version 1.0.0 -current_version 1.0.0  \
 -install_name "@rpath/${MVK_DYLIB_NAME}"  \
@@ -19,6 +19,6 @@
 -iframework ${MVK_SYS_FWK_DIR}  \
 -framework Metal -framework IOSurface -framework IOKit -framework QuartzCore -framework Foundation \
 --library-directory ${MVK_USR_LIB_DIR} \
--lSystem  -lc++ \
+-lSystem -stdlib=${CLANG_CXX_LIBRARY} \
 -o "${MVK_BUILT_PROD_PATH}/${MVK_DYLIB_NAME}" \
 -force_load "${MVK_BUILT_PROD_PATH}/${MVK_PROD_NAME}.framework/${MVK_PROD_NAME}"
